### PR TITLE
MEN-6121: Switch to CMake

### PR DESCRIPTION
**Pull request by vpodzime**
### Description

Switches the project from Autotools to CMake and introduces a dependency on LMDB (and C++ bindings for it) to see how things work with CMake.

### CODE REVIEW

**Comment by vpodzime** [comment]
@mender-test-bot start pipeline

**Comment by vpodzime** [comment]
[`test-docker`](https://gitlab.com/Northern.tech/Mender/mender/-/jobs/3496833867) fails because I had to switch the container to Ubuntu 22.04 because of the LMDB C++ bindings and C++ 17 required and the repo manipulation fails

**Review by oleorhagen — CHANGES_REQUESTED** [review]
Nice!

**Review by oleorhagen — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -38,12 +38,12 @@
...
-*_test
+*_test*
 
 # Build artifacts
 /mender
-/mender-auth/men-authd
-/mender-update/men-updated
+/mender-auth/mender-auth
+/mender-update/mender-update
 /common/json/impl/boost/libjson_la-boost_json.lo
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply .gitignore:47]: Yeah, you forgot to remove it and change those names :stuck_out_tongue:

File: tests/Dockerfile.daemon [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-RUN apt update && apt install -y make git build-essential jq curl autotools-dev autoconf libtool
+RUN apt update && apt install -y cmake make git build-essential jq curl pkg-config liblmdb++-dev
 
 WORKDIR /cpp/src/github.com/mendersoftware/mender
 COPY ./ .
 
-RUN ./autogen.sh
-RUN ./configure --prefix=/mender-install
+RUN cmake .
 RUN make
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply tests/Dockerfile.daemon:12]: Not a difference here, but I run `cmake . && make` on my machine because `make` is an alias to `make -j -l10`.

**Comment by vpodzime** [comment]
I guess it should be noted here that CMake might require some extra work from us (compared to Autotools) when doing [system introspection](https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20System%20Introspection.html) (Autotools handle this automatically/implicitly).

**Comment by vpodzime** [comment]
> ** ci/gitlab ** — Pipeline failed on Gitlab

Don't know how to fix or even debug this:
```
Err:1 http://archive.ubuntu.com/ubuntu jammy InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Err:2 http://security.ubuntu.com/ubuntu jammy-security InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
Reading package lists...
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.
W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C
...
```
Works just fine locally (and so does for @oleorhagen).

**Comment by vpodzime** [comment]
> Don't know how to fix or even debug this:

Solved by not requiring Ubuntu 22.04

**Review by oleorhagen — APPROVED** [review]
Nice!

**Pull request merged**